### PR TITLE
Updated 'performance_insights_kms_key_id' attribute to handle default CMK usage

### PR DESCRIPTION
**Pull request by mayank0202**
Tested on poc account

### CODE REVIEW

File: example/main.tf [hunk 1]
```diff
@@ -47,6 +47,8 @@ module "aurora" {
   aurora_allowed_cidr_blocks            = [data.aws_vpc.vpc.cidr_block]
   performance_insights_enabled          = true
   performance_insights_retention_period = 7
+  performance_insights_kms_key_id       = data.aws_kms_alias.aurora_cluster_kms_alias.target_key_arn
```
Comment (vijay-stephen) [inline example/main.tf:50]: use kms arn instead of alias

File: main.tf [hunk 2]
```diff
@@ -131,9 +131,9 @@ resource "random_password" "rds_db_admin_password" {
   }
 }
 
-data "aws_kms_alias" "rds" {
-  name = "alias/aws/rds"
-}
+# data "aws_kms_alias" "rds" {
+#   name = "alias/aws/rds"
```
Comment (vijay-stephen) [inline main.tf:135]: Please remove it , if its not required.

File: main.tf [hunk 3]
```diff
@@ -172,7 +172,7 @@ module "aurora_cluster" {
   rds_monitoring_interval = 30
 
   performance_insights_enabled          = var.performance_insights_enabled
-  performance_insights_kms_key_id       = var.performance_insights_enabled ? data.aws_kms_alias.rds.id : ""
+  performance_insights_kms_key_id       = var.performance_insights_enabled ? coalesce(var.performance_insights_kms_key_id, aws_kms_key.aurora_cluster_kms_key[0].arn) : ""
```
Comment (vijay-stephen) [inline main.tf:175]: I am not sure whether KMS key has to be created inside the module.

Comment (mayank0202) [reply main.tf:175]: Kms resource is written in inside the module already so if we remove then we can change it to default kms but currently its written 2 years ago

Comment (vijay-stephen) [reply main.tf:175]: is the create kms key being used any where?

Comment (mayank0202) [reply main.tf:175]: `aurora_cluster_kms_key` is not being used anywhere except i am using this in that performance attribute condition which can be removed if we are planning to remove the resource of aurora_cluster_kms_key and alias

@vijay-stephen  @jamescrowley321  @tsaucier-sf

File: example/main.tf (hunk 1 above)
Comment (mayank0202) [reply example/main.tf:50]: This is fetching the kms arn only @vijay-stephen

File: main.tf (hunk 2 above)
Comment (mayank0202) [reply main.tf:135]: If we remove the resurce of aurora kms key and alias then this would be used

File: example/main.tf (hunk 1 above)
Comment (mayank0202) [reply example/main.tf:50]: I got it , i will change the naming convention of this data lookup

File: example/data.tf [hunk 4]
```diff
@@ -32,3 +32,7 @@ data "aws_security_groups" "db_sg" {
     values = [data.aws_vpc.vpc.id]
   }
 }
+
+data "aws_kms_alias" "aurora_cluster_kms_arn" {
+  name = "alias/arc-poc-aurora-cluster-kms-key" # Replace with your KMS alias name
```
Comment (tsaucier-sf) [inline example/data.tf:37]: Make this a variable

File: main.tf (hunk 3 above)
Comment (tsaucier-sf) [reply main.tf:175]: Performance insight should probably have its own as you've added.

**Review by tsaucier-sf — CHANGES_REQUESTED** [review]

**Review by tsaucier-sf — APPROVED** [review]

**Review by bcrysler — APPROVED** [review]

**Pull request merged**